src/components/CytoscapeGraph/Layout/GroupCompoundLayout.ts

```diff
@@ -107,117 +110,119 @@ export default class GroupCompoundLayout {
+        };
+
+        const newStyles: OverridenStyles = {
+          shape: 'rectangle',
+          height: `${boundingBox.h}px`,
+          width: `${boundingBox.w}px`
+        };
+        // Saves a backup of current styles to restore them after we finish
+        parent.scratch(STYLES_KEY, backupStyles);
+        parent.addClass(COMPOUND_PARENT_NODE_CLASS);
+        // (1.b) Set the size
+        parent.style(newStyles);
       });
-    });
-    // (1.d) Remove all child nodes from parents (and their edges).
-    const removedElements = this.cy.remove(children);
-
-    // Ensure we only touch the requested elements and not the whole graph.
-    const layoutElements = this.cy.collection().add(this.elements).subtract(removedElements).add(syntheticEdges);
-
-    // Before running the layout, reset the elements positions.
-    // This is not absolutely necessary, but without this we have seen some problems with
-    //  `cola` + firefox + a particular mesh
-    layoutElements.position({ x: 0, y: 0 });
-
-    const layout = this.cy.layout({
-      // Create a new layout
-      ...this.options, // Sharing the main options
-      name: realLayout, // but using the real layout
-      eles: this.cy.elements(), // and the current elements
-      realLayout: undefined // We don't want this realLayout stuff in there.
-    });
 
-    // (2) Add a one-time callback to be fired when the layout stops
-    layout.one('layoutstop', _event => {
-      // If we add any children back, our parent nodes position are going to take the bounding box's position of all
-      // their children. Before doing it, save this position in order to add this up to their children.
-      parents.each(parent => {
-        parent.scratch(PARENT_POSITION_KEY, { ...parent.position() }); // Make a copy of the position, its an internal data from cy.
+      // (1.c) Add synthetic edges for every edge that touches a child node.
+      let syntheticEdges = this.cy.collection();
+      children.each(child => {
+        child.connectedEdges().each(edge => {
+          // (1.c) Create synthetic edges.
+          const syntheticEdge = this.syntheticEdgeGenerator.getEdge(edge.source(), edge.target());
+          if (syntheticEdge) {
+            syntheticEdges = syntheticEdges.add(this.cy.add(syntheticEdge));
+          }
+        });
+      });
+      // (1.d) Remove all child nodes from parents (and their edges).
+      const removedElements = this.cy.remove(children);
+
+      // Ensure we only touch the requested elements and not the whole graph.
+      const layoutElements = this.cy.collection().add(this.elements).subtract(removedElements).add(syntheticEdges);
+
+      // Before running the layout, reset the elements positions.
+      // This is not absolutely necessary, but without this we have seen some problems with
+      //  `cola` + firefox + a particular mesh
+      layoutElements.position({ x: 0, y: 0 });
+
+      const layout = this.cy.layout({
```

Comment:
This is done for the top-level nodes IIRC, so we are only using the main layout with the top elements.